### PR TITLE
update sd export for having SDPA

### PR DESCRIPTION
**Pull request by eaidova**

### CODE REVIEW

**Review by sammysun0711 — CHANGES_REQUESTED** [review]

File: llm_bench/python/convert.py [hunk 1]
```diff
@@ -363,240 +326,240 @@ def convert_seq2seq(args):
...
-                tokenizer.save_pretrained(ov_int8_dir.joinpath("tokenizer"))
-
-            tokenizer_2 = getattr(model, "tokenizer_2", None)
-            if tokenizer_2 is not None:
-                tokenizer_2.save_pretrained(ov_int8_dir.joinpath("tokenizer_2"))
-
-            model.save_config(ov_int8_dir)
-
     del model
     gc.collect()
```
[the remark this answers is not in the document]
Comment (eaidova) [reply llm_bench/python/convert.py]: moved

File: llm_bench/python/convert.py [hunk 2]
```diff
@@ -363,240 +326,240 @@ def convert_seq2seq(args):
...
+        wc_unet = compress_weights(pt_model.unet)
+        wc_vae = compress_weights(pt_model.vae)
+        pt_model.text_encoder = wc_text_encoder
+        pt_model.unet = wc_unet
+        pt_model.vae = wc_vae
+        output = Path(args.output_dir) / PYTORCH_DIR / OV_DIR / PYTORCH_COMPRESS_WEIGHTS_DIR.format(args.precision)
+        convert_sd_common(pt_model, output, args)
         del pt_model
         gc.collect()
+        pt_model = StableDiffusionPipeline.from_pretrained(args.model_id)
```
[the remark this answers is not in the document]
Comment (eaidova) [reply llm_bench/python/convert.py]: I changed order for conversion model, now fp conversion happens before weights compression, so we do not need recreate model

File: llm_bench/python/convert.py [hunk 3]
```diff
@@ -363,240 +326,240 @@ def convert_seq2seq(args):
...
+            model_type="clip-text-with-projection",
+        )
+        onnx_config = onnx_config_constructor(pipeline.text_encoder_2.config, int_dtype=int_dtype, float_dtype=float_dtype)
+        models_for_export["text_encoder_2"] = (models_for_export["text_encoder_2"], onnx_config)
+
+    return models_for_export
+
+
+def convert_sd_common(pipeline, output_dir, args, tiny_vae=False):
+    models_and_onnx_configs = _get_submodels_for_export_stable_diffusion(pipeline)
```
[the remark this answers is not in the document]
Comment (eaidova) [reply llm_bench/python/convert.py]: fixed, thanks

File: llm_bench/python/convert.py [hunk 4]
```diff
@@ -363,240 +326,240 @@ def convert_seq2seq(args):
...
-    log.info(f"Conversion total time {end - start}s")
-
-    if is_fp16(args):
-        model.half()
-    start1 = time.perf_counter()
-    model.save_pretrained(Path(args.output_dir) / PYTORCH_DIR / OV_DIR / args.precision)
-    end1 = time.perf_counter()
-    log.info(f"Serialization total time {end1 - start1}s")
+        output_dir = Path(args.output_dir) / PYTORCH_DIR / OV_DIR / args.precision
+    convert_sd_common(pt_model, output_dir)
```
[the remark this answers is not in the document]
Comment (eaidova) [reply llm_bench/python/convert.py]: fixed

File: llm_bench/python/convert.py [hunk 5]
```diff
@@ -363,240 +326,240 @@ def convert_seq2seq(args):
...
-    log.info(f"Conversion total time {end - start}s")
-
-    if is_fp16(args):
-        model.half()
-    start1 = time.perf_counter()
-    model.save_pretrained(Path(args.output_dir) / PYTORCH_DIR / OV_DIR / args.precision)
-    end1 = time.perf_counter()
-    log.info(f"Serialization total time {end1 - start1}s")
+    output_dir = Path(args.output_dir) / PYTORCH_DIR / OV_DIR / args.precision
+    convert_sd_common(pt_model, output_dir)
```
[the remark this answers is not in the document]
Comment (eaidova) [reply llm_bench/python/convert.py]: fixed

File: llm_bench/python/convert.py [hunk 6]
```diff
@@ -275,23 +248,15 @@ def convert_seq2seq(args):
                 )
```
[the remark this answers is not in the document]
Comment (eaidova) [reply llm_bench/python/convert.py:248]: fixed, thanks

File: llm_bench/python/convert.py [hunk 7]
```diff
@@ -363,240 +326,240 @@ def convert_seq2seq(args):
...
-            if tokenizer_2 is not None:
-                tokenizer_2.save_pretrained(output.joinpath("tokenizer_2"))
+        pipeline.save_config(output_dir)
 
-            pt_model.save_config(output)
+        export_models(
+            models_and_onnx_configs=models_and_onnx_configs,
+            output_dir=output_dir,
+            output_names=files_subpaths,
+            compress_option="fp16" if args.precision == "FP16" else None,
```
[the remark this answers is not in the document]
Comment (eaidova) [reply llm_bench/python/convert.py]: fixed, thanks

**Comment by sammysun0711** [comment]
LGTM, thanks for your great support!

**Review by sammysun0711 — APPROVED** [review]

**Pull request merged**